tests/decodecorpus.c

```diff
@@ -662,6 +674,12 @@ static U32 generateSequences(U32* seed, frame_t* frame, seqStore_t* seqStore,
                           MIN(frame->header.windowSize,
                               (size_t)((BYTE*)srcPtr - (BYTE*)frame->srcStart))) +
                          1;
+                if (info.useDict && (RAND(seed) & 1)) {
+                    /* need to occasionally generate offsets that go past the start */
+                    /* we still need to be within the windowSize however */
```

Comment:
minor clarification : you don't need to respect `windowSize` as a max distance for a match within the dictionary.
In fact, there is no limit, save the size of dictionary itself.
However, the dictionary cease being accessible after regenerating `windowSize` bytes.